### PR TITLE
Events layer rendering 

### PR DESCRIPTION
**Pull request by Lin1404**

### CODE REVIEW

File: src/components/calendar/week/Weekday.css [hunk 1]
```diff
@@ -4,6 +4,8 @@
 }
 
 .dayBody {
+    position: absolute;
```
Comment (HaofengLiang) [inline src/components/calendar/week/Weekday.css]: Can you confirm if you need these?

File: src/components/calendar/week/Weekday.js [hunk 2]
```diff
@@ -31,39 +20,31 @@ export default function WeekDay({ day, onEventSelect, isHeader }) {
...
             description:'',
             priority:0
-        };
-
-        const event = events?.find(event =>
-            event.startDateTime.isSame(dateTime) || event.startDateTime.isBetween(dateTime, moment(dateTime).add(minsPerSection, 'minutes'))
-        );  
+        }; 
         
         let rowSpan = 1;
```
Comment (HaofengLiang) [inline src/components/calendar/week/Weekday.js]: This might not be needed anymore.

File: src/components/calendar/week/Weekday.js [hunk 3]
```diff
@@ -31,39 +20,31 @@ export default function WeekDay({ day, onEventSelect, isHeader }) {
...
-                        {event && <Event event={event} /> }
-                    </ListItemButton>
+                </ListItem> 
+                :
+                <ListItem key={dateTime.format('MM-DD-yyyy-hh-mm-A') + "-list-item"} className='datetimeItem' style={{ height: heightPerRowInVH + "vh" }}>
+                    <ListItemButton onClick={() => onEventSelect(newEvent)} className="datetimeButton"></ListItemButton>
                 </ListItem >
         );
 
         i += rowSpan;
```
Comment (HaofengLiang) [inline src/components/calendar/week/Weekday.js]: This might not be needed anymore.

File: src/components/event/Events.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import React from "react";
+import { ListItem, ListItemButton } from '@mui/material';
+import moment from "moment";
+import { useSelector } from "react-redux";
+import "./Events.css";
+
+export default function Events({ day, onEventSelect }) {    
+    const events = useSelector((state) => state.events.value.filter(item => item.startDateTime.isSame(day, "day")));
+
+    const positionMapping = events.map(event => {
```
Comment (HaofengLiang) [inline src/components/event/Events.js]: "positionMapping" doesn't sound like a good name. "eventItems" or "eventDisplayItems" are more meaningful examples.

File: src/components/event/Events.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        const startDateTime = moment(event.startDateTime);
+        const endDateTime = moment(event.endDateTime);
+        const positionDiff = endDateTime.diff(startDateTime, 'minutes')/ 1440 * 240;
```
Comment (HaofengLiang) [inline src/components/event/Events.js]: nit: there is a format issue here.

(A nit is a small, insignificant issue spotted during a code review process that doesn't have a major impact on the overall quality of the code. )

File: src/components/event/Events.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import { ListItem, ListItemButton } from '@mui/material';
+import moment from "moment";
+import { useSelector } from "react-redux";
+import "./Events.css";
+
+export default function Events({ day, onEventSelect }) {    
+    const events = useSelector((state) => state.events.value.filter(item => item.startDateTime.isSame(day, "day")));
+
+    const positionMapping = events.map(event => {
+        const startDateTime = moment(event.startDateTime);
```
Comment (HaofengLiang) [inline src/components/event/Events.js]: Can you try if removing lines 12 and 13 makes any difference?

File: src/components/event/Events.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+                        <div className="eventTime">{event.startDateTime.format('hh:mm A')} - {event.endDateTime.format('hh:mm A')}</div>
+                        <div className="eventDescription">{event.description}</div>
+                    </div> 
+                </ListItemButton>
+            </ListItem >
+        )
+    })
+
+    return(
+        <div>
```
Comment (HaofengLiang) [inline src/components/event/Events.js]: Is this div redundant?

**Review by HaofengLiang — DISMISSED** [review]

File: src/components/event/Events.js (hunk 5 above)
Comment (weixiang237) [reply src/components/event/Events.js]: @Lin1404 I think what Feng means here is to fix the formatting of the highlighted codes.

here's the link to see how to set Vscode to 'Format on save' 

https://stackoverflow.com/questions/39494277/how-do-you-format-code-on-save-in-vs-code

**Review by weixiang237 — DISMISSED** [review]

**Review by HaofengLiang — APPROVED** [review]

**Pull request merged**